app/layout.tsx

```diff
@@ -1,17 +1,22 @@
+'use client';
 import "./globals.css";
-import React from "react";
+import React, {useState} from "react";
 import SideBar from "@/app/SideBar";
-import Home from '@/app/page'
+import userContext from "@/app/userContext";
 
-export default function RootLayout({children,}:{ children: React.ReactNode; }) {
-  return (
-    <html lang="en">
-      <body>
-      <main>
-          {children}
-          <SideBar/>
-      </main>
-      </body>
-    </html>
-  );
+export default function RootLayout({children,}: { children: React.ReactNode; }) {
+    const [user, setUser] = useState('');
+
+    return (
+        <userContext.Provider value={[user, setUser]}>
```

Comment:
try and convert this into a provider component that also keeps the state of the user and use just the provider as a wrapper for the app